sycl/test-e2e/Matrix/common.hpp

```diff
@@ -156,15 +162,26 @@ void matrix_copy(unsigned int rows, unsigned int cols, T *src, T *dst) {
   }
 }
 
-template <typename T1, typename T2, bool exact = false>
+template <Activation act = Activation::None, typename T1, typename T2,
+          bool exact = false>
 bool matrix_compare(unsigned int rows, unsigned int cols, T1 *src, T2 *ref) {
   for (int i = 0; i < rows; i++) {
     for (int j = 0; j < cols; j++) {
       if constexpr (!exact && (std::is_same_v<T1, float> ||
                                std::is_same_v<T1, bfloat16> ||
                                (std::is_same_v<T1, double> &&
                                 std::is_same_v<T2, double>))) {
-        float diff = std::fabs(src[i * cols + j] - (T1)ref[i * cols + j]);
+        float diff = 0;
```

Comment:
If we want to introduce activation functions to Joint Matrix common test library, it should be separate function. We should not overload matrix_compare with additional logic.
But I'm not sure we actually need activation in SYCL E2E Joint Matrix tests. It doesn't seem to belong to Joint Matrix extension functionality and can be replaced with any other function (say x+2). Or am I missing something?

Reply:
I will try to break it down like you suggested in your previous comment. Separate the apply from the compare. I should be able to keep the compare as we have it today.